### ISSUE TITLE
ERROR: Calling NotifierModule.forRoot results in 'Error encountered resolving symbol values statically'

### ISSUE THREAD

**Issue by harryhahn2** [dominique-mueller/angular-notifier#5]
in app.modules.ts : NotifierModule.forRoot({})  throws an error
when i build the angular2-cli project.

```
ERROR in Error encountered resolving symbol values statically. Calling function 'NotifierConfig', function calls are not supported. Consider replacing the function or lambda with
 a reference to an exported function, resolving symbol NotifierModule.forRoot in C:/Entwicklung/Projekte/marenta/marenta/node_modules/angular-notifier/src/notifier.module.d.ts, r
esolving symbol AppModule in C:/Entwicklung/Projekte/marenta/marenta/src/app/app.module.ts, resolving symbol AppModule in C:/Entwicklung/Projekte/marenta/marenta/src/app/app.modu
le.ts
```

**Comment by dominique-mueller** [comment]
Just published v1.0.3, which (hopefully) should fix the issue you described. For details regarding why that problem even existed I can also refer to two Angular issues:

- https://github.com/angular/angular/issues/11262
- https://github.com/angular/angular/issues/10789

Could you run an `npm update` and verify that everything works now as expected?

**Comment by mashhoodr** [comment]
@dominique-mueller Im on 1.0.3 and getting the error when using the latest CLI version (@angular/cli RC.0)

```
ERROR in Error encountered resolving symbol values statically. Could not resolve type NotifierOptions (position 116:37 in the original .ts file), resolving symbol NotifierConfig in /Users/.../node_modules/angular-notifier/src/models/notifier-config.model.d.ts
```

**Comment by dominique-mueller** [comment]
Hey there, quiet an interesting error you get there. I tried to reproduce it, thus created a quick Angular app with the latest version of 2.4.x, together with the latest version of the Angular CLI (RC.0).

I tried a whole bunch of constellations, such as
- building for development
- building for production, both with and without AoT
- using the default configuration of angular-notifier, or passing in custom ones.

However, _**I'm not able to get the issue you described above**_.

Do you have more information about the issue, or maybe are you able to create a Plunkr showing it? Which exact Angular version are you using (latest is 2.4.8)?

**Comment by mashhoodr** [comment]
Hello @dominique-mueller 

Thanks for the quick reply!

I removed my node_modules folder and reinstalled everything, so my current angular version is at 2.4.7. This issue does not happen for me if I turn AOT off, or do just `ng build`. Do you have any other recommendation of things I can check? I will also try creating a Plunkr to reproduce this issue.

Also I was going through the library code and I found in the components you have private variables (notifications and config) which are also being used in the html template - not sure if that was related.

Cheers.

**Comment by dominique-mueller** [comment]
The problem got described in more details in issue #7, and fixed with #8.

**Comment by mashhoodr** [comment]
@dominique-mueller thanks, I will try out the latest version.

**Comment by mashhoodr** [comment]
@dominique-mueller I can confirm the problem has been fixed for me with version 1.0.4 thanks a bunch!
